src/Compilers/Core/Portable/SourceGeneration/UserFunction.cs

```diff
@@ -30,9 +30,7 @@ internal static class UserFunctionExtensions
                 {
                     return userFunction(input, token);
                 }
-#pragma warning disable CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete; tracked by https://github.com/dotnet/roslyn/issues/58375
-                catch (Exception e) when (FatalError.ReportIfNonFatalAndCatchUnlessCanceled(e, token))
-#pragma warning restore CS0618 // ReportIfNonFatalAndCatchUnlessCanceled is obsolete
+                catch (Exception e) when (!ExceptionUtilities.IsCurrentOperationBeingCancelled(e, token))
```

Comment:
❔ Are we still going to get NFW reporting if/when these occur inside the IDE

Reply:
As a separate change we're going to add an API to the GeneratorDriver to let us hook and report these, which is similar to how we do this for analyzers. Having this via an internal only API that's mixed in for fault reporting is a bit funky.

Reply:
@chsienki Do we have a tracking bug to actually do that?